tests/helpers/test_modifier.py

```diff
@@ -0,0 +1,148 @@
+from typing import Union
+from unittest.mock import Mock, mock_open, patch
+
+import pytest
+from flow.record import Record
+from flow.record.fieldtypes import digest, path
+
+from dissect.target import Target
+from dissect.target.exceptions import FileNotFoundError, IsADirectoryError
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record_modifier import (
+    Modifier,
+    ModifierFunc,
+    get_modifier_function,
+)
+from tests.helpers.test_hashutil import HASHES
+
+
+@pytest.fixture
+def hash_function() -> ModifierFunc:
+    return get_modifier_function(Modifier.HASH)
+
+
+@pytest.fixture
+def resolve_function() -> ModifierFunc:
+    return get_modifier_function(Modifier.RESOLVE)
+
+
+@pytest.mark.parametrize(
+    "test_input, expected_records",
+    [
+        ({"name": path}, 2),
+        ({"name": path, "test": path}, 3),
+        ({"name": path, "test": str}, 2),
+    ],
+)
+@patch("flow.record.Record")
+def test_hash_path_records_with_paths(
+    record: Record,
+    hash_function: ModifierFunc,
+    target_win: Mock,
+    test_input: dict[str, Union[type[path], type[str]]],
+    expected_records: int,
+) -> None:
+    record._desc.name = "test"
+    record._field_types = test_input
+
+    record_names = [key for key, value in test_input.items() if value is path]
+
+    with (
+        patch.object(TargetPath, "open", mock_open(read_data=b"")),
+        patch("dissect.target.helpers.record_modifier.common", return_value=HASHES),
+    ):
+        hashed_record = hash_function(target_win, record)
+
+    assert hashed_record.name == "test"
+    assert len(hashed_record.records) == expected_records
+    assert hashed_record.records[0] == record
+
+    _record = hashed_record.records[1]
```

Comment:
```suggestion
    _record = hashed_record.records[1]
```

This looks like it is not being used.